### PR TITLE
add CRAFT training code

### PR DESCRIPTION
**Pull request by gmuffiness**
* A PR to add training code of CRAFT(Baek et al., 2019)
* https://github.com/JaidedAI/EasyOCR/issues/736#issuecomment-1138733029

### CODE REVIEW

**Comment by rkcosmos** [comment]
thanks, I'll test it and get back to you.

**Comment by rkcosmos** [comment]
hi, i have a few questions.

1. Is the reported score coming from training from scratch? Why do we need to download a pretrain in training step 2?
2. how many synthtext images do you use to produce the result? Is it from pre-generated 800,000? 
3. To produce a pretrain for SynthText + ICDAR2015 dataset (second row in readme), do you start from a checkpoint from SynthText only? (I saw `ckpt_path` in ic15_train.yaml. I guess it is from the one with 50,000 iter.)

Thanks.

**Comment by gmuffiness** [comment]
Thanks for the quick response.

1. First row in readme is a model trained with SynthText from scratch, and second row in readme loaded the first step model checkpoint that is trained for 50,000 iter with SynthText and start training for 25,000 iterations with SynthText and IC15. 

Training step 2 model is simply attached to show that model (trained with this training code) achieves performance comparable to papers. If you don't need it, you don't need to download it actually. If that confuses you a bit, is it better to delete it?

2. Yes, 800,000 images provided by the following link. https://www.robots.ox.ac.uk/~vgg/data/scenetext/

3. Yes. I trained only SynthText total 50,000 iter, but among them, the checkpoint with the best accuracy was uploaded. 
    I can change the best accuracy checkpoint to final checkpoint (50,000 iter) or add it. There is no significant difference in performance between the two. What would be better for me to do?

Feel free to ask if you have any more questions.

**Comment by rkcosmos** [comment]
Thanks for the explanation. 

I tried to clarify training section in readme file a bit. Please comment if it is accurate.

**Comment by gmuffiness** [comment]
Thanks for the supplementary explanation.

I changed the readme and add separate training script for SynthText dataset.
I think it will be easier to understand how to use now.

**Comment by rkcosmos** [comment]
Merged. 

Dependency bot reports a few security issues with pillow==8.2.0. Could you check if the code still work for higher pillow version? The bot suggest pillow==9.0.1.

**Comment by gmuffiness** [comment]
Thanks!

I checked that it works well in the higher pillow version. (pillow==9.0.1)

**Comment by VuDinhToan303** [comment]
Hello gmuffiness, 
I have tried to train the Craft model using your code. However, when training with my data, I encountered the following error: Error : operands could not be broadcast together with shapes (200,200) (330,0) 
On generating affinity map, strange box came out. (width: 0, height: 330)
This error came from gaussian.py, how can i fix that?Thank you
